### DISCUSSION TITLE
Feature request: Streaming JSON lines into DB

### DISCUSSION THREAD

**Discussion by v1gnesh** [duckdb/duckdb#8776] [Ideas]
Firstly, thank you for your excellent, excellent work over the years, and for caring enough to make good software.

In Rust land, say there's buffered futures that will become `Vec<u8>` when `await`ed, where each `Vec<u8>` from the `Stream` represents a JSON object's bytes. 
How can I stream this right into a DuckDB... maybe batch/chunk 1000 of them together to save on writes for every Stream Item?

Is this possible today? If not, wouldn't this be good to have for streaming conversion into Duck, without an interim stop (JSON/CSV/etc. saved to disk)?
I'm hoping to avoid doing `INSERT INTO tbl SELECT * FROM read_json_auto('file')`, because a language-native way may fit better.

**Comment by paulrutter** [comment]
I had the same question for the Node Api, where i found a performant solution using Apache Arrow as intermediate format (which doesn't require making a copy of the data).

Maybe that will work for Rust as well?

See https://github.com/duckdb/duckdb/discussions/8466#discussioncomment-6907765

**Reply by v1gnesh** [reply]
Glad that you found a solution for your case.
I'm afraid this won't work for Rust, as duckdb-rs's bindings are skeletal at the moment.
Ideally, I want to avoid the interim JSON stop itself.
Unfortunately, I don't know when duckdb-rs will get some love, that it allows the appender to work directly with Rust data types.
